bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1053,7 +1053,13 @@ impl<'a> CircuitInputStateRef<'a> {
         };
         let gas_refund = geth_step.gas.0 - memory_expansion_gas_cost - code_deposit_cost;
 
-        let caller_gas_left = geth_step_next.gas.0 - gas_refund;
+        // revert also make call.is_success = false, check for only RETURN in create for
+        // oog code store maybe better check here.
+        let caller_gas_left = if !call.is_success && geth_step.op == OpcodeId::RETURN {
```

Comment:
should polish the grammar here.. a bit difficult to understand..

Reply:
should explain more.     if a RETURN opcode fails, there can be 2 reasons, out of gas or stack error, both will cost all the remaining gas so there will be no refund.   While for REVERT, it will not consume all the remaining gas.

Reply:
Eh will other opcodes except return and revert reach here？if so we should check the opcode not revert instead of checking it is return

Reply:
I am thinking even the revert opcode itself can be run out of gas due to memory expansion ... So the calculation of refund may panic？you can directly in this pr add a revert out of gas test case to see what happens..

Reply:
And no creation return can run out of gas due to memory expansion too..    It is not difficult i prefer we fix them all in bus mapping side here all at once and finish gadget later

Reply:
it is not RETURN error,  but create related errors(invalid creation code, max code size exceed, codestore oog) are detected when executing op code is RETURN. which is from https://github.com/scroll-tech/zkevm-circuits/blob/scroll-stable/bus-mapping/src/circuit_input_builder/input_state_ref.rs#L1182 
the RETURN is the last byte code of init code in create run. after all init code runs, then check if error happens.
it is special since the error step we are checking is not create itself.